mne/time_frequency/tests/test_spectrum.py

```diff
@@ -386,3 +387,55 @@ def test_spectrum_kwarg_triaging(raw):
         raw.plot_psd(axes=axes)
     # `ax` is the correct legacy param name
     raw.plot_psd(ax=axes)
+
+
+def test_spectrumarray_raw(raw):
+    """Test SpectrumArray for Raw-derived spectra."""
+    spect = raw.compute_psd()
+    spect2 = SpectrumArray(
+        data=spect.get_data(),
+        freqs=spect.freqs,
+        info=spect.info,
+    )
+    assert_array_equal(spect.get_data(), spect2.get_data())
+    assert_array_equal(spect.freqs, spect2.freqs)
+
+
+def test_epochsspectrumaray(epochs):
+    """Test EpochsSpectrumArray for Epochs-derived spectra."""
+    spect = epochs.compute_psd()
+    spect2 = EpochsSpectrumArray(
+        data=spect.get_data(),
+        freqs=spect.freqs,
+        info=spect.info,
+        events=epochs.events,
+        event_id=epochs.event_id,
+    )
+    assert_array_equal(spect.get_data(), spect2.get_data())
+    assert_array_equal(spect.freqs, spect2.freqs)
+
+
+def test_plot_spectrumaray(raw, epochs):
+    """Test SpectrumArray and EpochsSpectrumArray plotting."""
+    spectra = list()
+    spect_orig = raw.compute_psd()
+    spectra.append(
+        SpectrumArray(
+            data=spect_orig.get_data(),
+            freqs=spect_orig.freqs,
+            info=spect_orig.info,
+        )
+    )
+    spect_orig = epochs.compute_psd()
```

Comment:
I would test the plotting in the test functions above to avoid the recomputation of the psd which is not free or use pytest fixture

Reply:
opened #11811